risc0/circuit/recursion/src/metal.rs

```diff
@@ -63,13 +63,23 @@ impl<MH: MetalHash> CircuitHal<MetalHal<MH>> for MetalCircuitHal<MH> {
     ) {
         const EXP_PO2: usize = log2_ceil(INV_RATE);
         let domain = steps * INV_RATE;
-        let poly_mix =
-            MetalBuffer::copy_from(&self.hal.device, self.hal.cmd_queue.clone(), &[poly_mix]);
+        let poly_mix = MetalBuffer::copy_from(
+            "poly_mix",
+            &self.hal.device,
+            self.hal.cmd_queue.clone(),
```

Comment:
IIRC the `cmd_queue` is ref-counted (basically it's actually a Objective-C thing under the hood). I don't think there's a way to avoid it but it should cost much.